### PR TITLE
[IMP]Adds function amount_to_text for spanish text numbers.

### PR DESCRIPTION
**Pull request by sasprilla**
Description of the issue/feature this PR addresses:

Replication, translation and settings of (amount_to_text_en)

Current behavior before PR:
Openerp not convert numbers to text in Spanish

Desired behavior after PR is merged:
convert numbers to text in Spanish
## 

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by matheo** [comment]
+1

**Comment by grottas** [comment]
Hi @odony @mart-e @Yenthe666 @dreispt ,

Is there any chance for this to be merged? Who should this be pointed to?

This is an important requirement for any Spanish language .

BTW (+1)

Regards!

File: openerp/tools/amount_to_text_es.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
```
Comment (grottas) [inline openerp/tools/amount_to_text_es.py]: Use global LICENSE/COPYRIGHT files, remove boilerplate text.

See: L1 and L2 from [here](https://github.com/odoo/odoo/blob/9.0/openerp/tools/amount_to_text.py).

File: openerp/tools/amount_to_text_es.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+'Veinte', 'Treinta', 'Cuarenta', 'Cincuenta', 'Sesenta', 'Setenta', 'Ochenta',
+'Noventa')
+denom = ('',
+         'Mil', 'Millón', 'Billón', 'Trillón', 'Cuatrillón',
+         'Quintillón', 'Sextillion', 'Septillion', 'Octillion', 'Nonillion',
+         'Decillion', 'Undecillion', 'Duodecillion', 'Tredecillion',
+         'Quattuordecillion',
+         'Sexdecillion', 'Septendecillion', 'Octodecillion', 'Novemdecillion',
+         'Vigintillion')
+
```
Comment (grottas) [inline openerp/tools/amount_to_text_es.py]: Is it necessary to add the **_es** sufix to **to_19**, **tens** and **denom**? as it is done for the French and Dutch languages?

File: openerp/tools/amount_to_text_es.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+def _convert_nn(val):
```
Comment (grottas) [inline openerp/tools/amount_to_text_es.py]: idem about the **_es** sufix

File: openerp/tools/amount_to_text_es.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    if val < 20:
+        return to_19[val]
+    for (dcap, dval) in ((k, 20 + (10 * v)) for (v, k) in enumerate(tens)):
+        if dval + 10 > val:
+            if val % 10:
+                return dcap + '-' + to_19[val % 10]
+            return dcap
+
+
+def _convert_nnn(val):
```
Comment (grottas) [inline openerp/tools/amount_to_text_es.py]: idem about the _es sufix

File: openerp/tools/amount_to_text_es.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                if l > 1:
+                    ret = _convert_nnn(l) + ' ' + denom[didx] + 'es de '
+                else:
+                    ret = _convert_nnn(l) + ' ' + denom[didx] + ' de '
+            if r > 0:
+                ret = ret + ', ' + spanish_number(r)
+            return ret
+
+
+def amount_to_text(number, currency):
```
Comment (grottas) [inline openerp/tools/amount_to_text_es.py]: idem about the **_es** sufix

File: openerp/tools/amount_to_text_es.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        end_word = spanish_number(int(list[1]))
+
+    return ' '.join(filter(None, [(start_word or end_word), units_name]))
+
+
+# -------------------------------------------------------------
+# Generic functions
+# -------------------------------------------------------------
+
+_translate_funcs = {'es': amount_to_text}
```
Comment (grottas) [inline openerp/tools/amount_to_text_es.py]: idem about the **_es** sufix

File: openerp/tools/amount_to_text_es.py [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+def amount_to_text(nbr, lang='es', currency='Pesos'):
+    if not _translate_funcs.has_key(lang):
+        lang = 'es'
+    return _translate_funcs[lang](abs(nbr), currency)
+
+
+if __name__ == '__main__':
+    from sys import argv
+
+    lang = 'nl'
```
Comment (grottas) [inline openerp/tools/amount_to_text_es.py]: nl or **es**?

File: openerp/tools/amount_to_text_es.py [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+
+# -------------------------------------------------------------
+# Generic functions
+# -------------------------------------------------------------
+
+_translate_funcs = {'es': amount_to_text}
+
+
+def amount_to_text(nbr, lang='es', currency='Pesos'):
```
Comment (grottas) [inline openerp/tools/amount_to_text_es.py]: Is there any way to do this more generic? Pesos is the currency for some countries, but not for all Spanish speaking countries.

**Comment by mart-e** [comment]
We were already talking about this in #1562 (2 years ago), the idea was we don't really want to add more languages in a system that doesn't work very well. Instead of this unmaintable code, we were looking for a library like num2words but nothing was decided in the end.

cc @antonylesuisse

**Comment by dreispt** [comment]
@mart-e If you provide an extensible mechanism for this, we could have community modules for this.

**Comment by grottas** [comment]
@mart-e It seems that you agree with the [num2words](https://github.com/savoirfairelinux/num2words) library.

So, is the best solution for this? Is that a method that Odoo SA can consider to get implemented?

Anyway, I tracked the amount_to_text issues and there is only that open issue related to the French language. English and Dutch languages seems to be working well, and also this Spanish PR, so I don't get why this code is unmaintainable.

While a decision is taken (now it's two years in a halt), can this be the solution for the Spanish language?

@dreispt Does the num2words library could be the extensible mechanism you are asking for? Can that be implemented as a community module? or maybe this PR can be implemented in OCB?

cc @elmonitor

**Comment by dreispt** [comment]
@grottas The num2words lib makes the extension mechanism unnecessary.
Odoo would just use it, and to add additional languages you just have to contribute to the lib (that is maintained by a well known Odoo integrator, BTW).

@mart-e @antonylesuisse Is adding this lib an option?

**Comment by mart-e** [comment]
Yes, we have opened a R&D task to use num2words so no extension mechanism is necessary.
But still discussing the best way, no coding yet :)

**Comment by dreispt** [comment]
Great to hear that!
At first sight is seems like a simple change...
I'm not sure if num2words is on Pypi, but if not we can open an Issue there.

**Comment by antonylesuisse** [comment]
Yes we will use num2word with a fall back to empty string if the lib cannot be located. PR are weclomed.

**Comment by mart-e** [comment]
Hello

Good news, num2words [has been added into Odoo](https://github.com/odoo/odoo/commit/e0be5bb11cd857708d57cc757a54eed044b76a4b) (since saas-14) !

Pull requests welcome to clean the remaining calls to the bad amount_to_text to convert them to num2words.

**Pull request closed without merging**